index.js

```diff
@@ -50,28 +51,65 @@ function Filter(inputTree, options) {
   this._destFilePathCache = Object.create(null);
 }
 
+Filter.prototype._resetStats = function() {
+  this._stats = {
+    paths: 0,
+    stale: 0,
+    mkdirp: 0,
+    mkdirpCache: 0,
+    processed: 0,
+    unprocessed: 0,
+    hits: 0,
+    miss: 0,
+    prime: 0,
+    hash: 0,
+    processedTime: 0
+  };
+};
+
 Filter.prototype.build = function build() {
   var self = this;
+  var start = new Date();
+  this._resetStats();
   var srcDir = this.inputPaths[0];
   var destDir = this.outputPath;
-  var paths = walkSync(srcDir);
+  var paths = walkSync(srcDir).filter(Boolean);
+
+  self._stats.paths = paths.length;
+  self._stats.walkSyncTime = new Date() - start;
 
   this._cache.deleteExcept(paths).forEach(function(key) {
+    self._stats.stale++;
     fs.unlinkSync(this.cachePath + '/' + key);
   }, this);
 
+  self._stats.mkdirp += mkdirpBulk.sync(paths.map(function(p) {
+    return self.outputPath + '/' + p;
+  }));
+
+  if (this._cacheDirsPrimed === undefined) {
```

Comment:
> so this would only seem to run once.

Ya the goal is for it to only run exactly once.

The variable name was written in haste :P